Server/bank/transaction/src/main/java/com/cashmanager/server/transaction/controller/TransactionController.java

```diff
@@ -0,0 +1,98 @@
+package com.cashmanager.server.transaction.controller;
+
+import com.cashmanager.server.account.service.AccountService;
+import com.cashmanager.server.account.service.PaymentMethodService;
+import com.cashmanager.server.common.dto.PaymentMethodDto;
+import com.cashmanager.server.common.dto.TransactionDto;
+import com.cashmanager.server.common.enumeration.TransactionStatus;
+import com.cashmanager.server.database.entity.PaymentMethod;
+import com.cashmanager.server.database.entity.Transaction;
+import com.cashmanager.server.database.entity.TransactionLog;
+import com.cashmanager.server.database.enumeration.LogSeverity;
+import com.cashmanager.server.database.mapper.PaymentMethodMapper;
+import com.cashmanager.server.database.mapper.TransactionMapper;
+import com.cashmanager.server.database.repository.AccountRepository;
+import com.cashmanager.server.database.repository.TransactionLogRepository;
+import com.cashmanager.server.database.repository.TransactionRepository;
+import com.cashmanager.server.transaction.model.TransactionHolder;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.time.LocalDateTime;
+import java.util.Optional;
+
+/**
+ * Controller in charge of dispatching request for returning the validation of a bank transaction
+ */
+@RestController
+public class TransactionController {
+
+    private TransactionLogRepository transactionLogRepository;
+    private TransactionRepository transactionRepository;
+    private AccountRepository accountRepository;
+    private PaymentMethodService paymentMethodService;
+    private AccountService accountService;
+
+    /**
+     * Starting point from mediator for payment method and transaction steps
+     * @param transactionHolder - Object containing payment method and amount of transaction
+     */
+    @PostMapping("/transaction")
+    public void checkTransaction(@RequestBody TransactionHolder transactionHolder) {
+
+        PaymentMethodDto paymentMethodDto = transactionHolder.getPaymentMethodDto();
+        TransactionDto transactionDto = TransactionHolder.setupTransactionDto(transactionHolder);
+
+        Optional<PaymentMethod> paymentMethod = paymentMethodService.checkPaymentMethodViability(paymentMethodDto, transactionDto);
+        if (paymentMethod.isPresent()) {
+            if (accountService.checkAccountValidity(accountRepository.findById(paymentMethod.get().getAccount().getId()), transactionDto)) {
```

Comment:
for this part, what i meant was to put the idAccount in the checkAccountValidity(), and in the method you use the accountRepository.findById(idAccount).
 ```suggestion
            if (accountService.checkAccountValidity(paymentMethod.get().getAccount().getId(), transactionDto)) {
```
To simplify this, you can even change the type return of the checkPaymentMethodViability to Optional&lt;UUID&gt; to return directly the account id and don't use the entity PaymentMethod in the controller.

Reply:
Thx for the review, I have applied your first suggestion.
Not the second, it's less understandable for the code on my view.